packages/myst-frontmatter/src/frontmatter/validators.ts

```diff
@@ -258,7 +266,11 @@ export function validateAndStashObject<T extends { id?: string; name?: string }>
       .digest('hex');
     warnOnDuplicate = false;
   }
-  if (!Object.keys(lookup).includes(value.id)) {
+  if (Object.keys(value).length === 1 && value.id) {
+    // This is a single object with an ID that may be defined later
+    // Only assign if it is not already defined
+    lookup[value.id] ??= value;
+  } else if (!Object.keys(lookup).includes(value.id)) {
```

Comment:
This is the only major change, if there is an object `{ id: 'ubc' }` coming in, we don't want to overwrite anything, and in the initial stages of validation we also don't want to warn. This is a linking action.